src/components/DOW/AnticipatedDurationandUsage.vue

```diff
@@ -129,9 +128,19 @@ export default class AnticipatedDurationandUsage extends Vue {
   // when user selects "NO", pre-select base period
   @Watch("_entireDuration")
   public entireDurationChanged(newVal: string): void {
-    this._selectedPeriods = newVal === "NO" && this.availablePeriodCheckboxItems[0].value !== "" 
-      ? [this.availablePeriodCheckboxItems[0].value]
-      : [];
+    if (this.availablePeriodCheckboxItems.length>0){
+      this._selectedPeriods = newVal === "NO" && this.availablePeriodCheckboxItems[0].value !== "" 
+        ? [this.availablePeriodCheckboxItems[0].value]
+        : [];
+    }
+
+    if (newVal === "NO"){
+      this.periodCheckboxRules.push(
+        this.$validators.required('Please select at least one base or option period.')
+      )
+    } else {
+      this.periodCheckboxRules = [];
+    }
```

Comment:
could do the above in a ternary with fewer lines of code:
```
this.periodCheckboxRules = newVal === "NO" 
    ? [this.$validators.required('Please select at least one base or option period.')]
    : [];
```

Reply:
done